pkg/miscellaneous/miscellaneous.go

```diff
@@ -462,5 +464,39 @@ func GetInitialClusterState(ctx context.Context, logger logrus.Entry, clientSet
 		return ClusterStateExisting
 	}
 
-	return ClusterStateNew
+	return ""
+}
+
+// DoPromoteMember promotes a given learner to a voting member.
+func DoPromoteMember(ctx context.Context, member *etcdserverpb.Member, cli etcdClient.ClusterCloser, logger *logrus.Logger) error {
+	memPromoteCtx, cancel := context.WithTimeout(ctx, brtypes.DefaultEtcdConnectionTimeout)
+	defer cancel()
+
+	//Member promote call will succeed only if member is in sync with leader, and will error out otherwise
+	_, err := cli.MemberPromote(memPromoteCtx, member.ID)
+	if err == nil {
+		//Member successfully promoted
+		logger.Infof("Member %v with [ID: %v] has been promoted", member.GetName(), strconv.FormatUint(member.GetID(), 16))
+		return nil
+	} else if errored.Is(err, rpctypes.Error(rpctypes.ErrGRPCMemberNotLearner)) {
+		//Member is not a learner
```

Comment:
Is there a different way to ascertain that the member is a voting member of the cluster? If at all in the future, etcd community introduces a new kind of member which is neither `learner` nor `voting member`, then this check will not work.

Reply:
we are currently using `etcd v3.4.13` and we don't see any plan to move  `v3.5.x` or `v3.6.x` anytime soon right ?
And even in etcd `v3.5.x` or `3.6.x` I don't see any new concept of new kindof member. so spending time on something which will not come anytime soon (for almost 2+ yrs) ... seems very unproductive to me right now.